runtime/compiler/control/CompilationThread.cpp

```diff
@@ -2596,6 +2618,60 @@ void TR::CompilationInfo::resumeCompilationThread()
    releaseCompMonitor(vmThread);
    }
 
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+void TR::CompilationInfo::prepareForCheckpoint()
+   {
+   J9JavaVM   *vm       = _jitConfig->javaVM;
+   J9VMThread *vmThread = vm->internalVMFunctions->currentVMThread(vm);
+
+   {
+   OMR::CriticalSection suspendCompThreadsForCheckpoint(getCompilationMonitor());
+
+   /* Set the checkpoint in progress flag. */
+   setCheckpointInProgress();
```

Comment:
Maybe we should have a fatal assert here that checkpoint cannot be in progress. This will avoid errors with two threads trying to start checkpointing in parallel.

Reply:
Addressed in latest force push.